### DISCUSSION TITLE
Stacks Node Developer Onboarding FAQ

### DISCUSSION THREAD

**Discussion by igorsyl** [stacks-network/stacks-core#3642] [Q&A]
1. Can multiple bootstrap nodes be used and in what use case would this be optimal?
https://github.com/stacks-network/stacks-blockchain/blob/master/testnet/stacks-node/src/config.rs#L1782-L1794

2. What are deny nodes?
https://github.com/stacks-network/stacks-blockchain/blob/master/testnet/stacks-node/src/config.rs#L1796-L1805

3. Where can the custom address for receiving coinbase transaction rewards be configured? Is it parsed when committing the rewarding block with RewardSetInfo or where is it? https://github.com/stacks-network/stacks-blockchain/blob/feat/2.1.0.0.0-rc2/testnet/stacks-node/src/neon_node.rs#L1733
Igor: Degens want to know how to configure the Stacks coinbase address (pay to contract).
The information will be encoded in the Commit op

5. Related to keychain file operations, multiple microblocks in the same block would have the same block_height and salt. Would the keypair be the same as well?
https://github.com/stacks-network/stacks-blockchain/blob/master/testnet/stacks-node/src/keychain.rs#L112-L127
/// Generate a microblock signing key for this burnchain block height.
/// `salt` can be any byte string; in practice, it's the parent Stacks block's block ID hash.
pub fn make_microblock_secret_key( &self, block_height: u64, salt: &[u8])

6. Related to files management, from Cargo.toml the packages such as stx_genesis, clarity and stacks_common are present in folders where their paths are specified. For the stacks package there is no folder and it would at the repo level. 
stacks = { package = "blockstack-core", path = "../../." } 
What are the files that imports such as use stacks::address::AddressHashMode;,
use stacks::burnchains::BurnchainSigner; are coming from ?

7. In other rust projects the analyzer auto-display types on variable declarations, on hover displays a pop-up with docs information and when using ctrl+click it goes to instantiations. When trying it on this project it just says loading and nothing from the above happens. The project and the files are a lot bigger than the ones compared where it works, could this be the problem or could it be something else?

**Comment by igorsyl** [comment]
From @wileyj 

1. yes, multiple nodes was the default for a while (currenlt load balanced i believe in a recent PR). if a bootstrap node is down/slow to respond, another known bootstrap node can be used vs waiting for the default. 
2. deny_nodes apears to be a blocklist of sorts, but ludo/aaron/brice can correct me if i'm incorrect

**Comment by obycode** [comment]
6. `rust-analyzer` is the component responsible for this. It sometimes trips up on large projects like this. Sometimes reloading the project helps, but sometimes, it doesn't. I haven't found a reliable reason to clarify when it works and when it doesn't (though I haven't really investigated).

**Reply by BowTiedDeployer** [reply]
6.Thanks to @igorsyl we found that Visual Studio Code has this limitation regarding bigger projects and `rust-analyzer` can't index them properly. We set up CLion instead and all the references work.    

 5.Because referencing is working now, we were able to successfully get from what files `use stacks::address::AddressHashMode`,  `use stacks::burnchains::BurnchainSigner;` and others were imported.

**Reply by igorsyl** [reply]
I ❤️ CLion

**Comment by BowTiedDeployer** [comment]
3. The address is not implemented right now. We'll create a commit for it and pull request it to the `stacks-blockchain` repo. It will be configured like the other fields (seed, local_peer_seed, rpc_bind)  and will be configurable using `reward_address=principal_address/smart_contract_address`

[Track development progress](https://github.com/stacks-degens/mining-coordinator/issues/1) and [forked repo](https://github.com/stacks-degens/stacks-blockchain).

**Reply by BowTiedDeployer** [reply]
it was completed in this [commit](https://github.com/stacks-degens/stacks-blockchain/commit/608eabff8d92003ac5d7ce3df5a3d33f599d8b1e)

it is configured as `block_reward_recipient=principal_address/smart_contract_address` 

https://github.com/stacks-degens/stacks-blockchain/blob/ebf6cf22aab880df6e129d216d446e80ba2f5049/testnet/stacks-node/src/config.rs#L1844
